File: app/src/main/java/com/example/easyupipayment/MainActivity.java
package com.example.easyupipayment;

import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ImageView;
import android.widget.RadioButton;
import android.widget.RadioGroup;
import android.widget.TextView;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

import com.shreyaspatil.EasyUpiPayment.EasyUpiPayment;
import com.shreyaspatil.EasyUpiPayment.listener.PaymentStatusListener;
import com.shreyaspatil.EasyUpiPayment.model.PaymentApp;
import com.shreyaspatil.EasyUpiPayment.model.TransactionDetails;

public class MainActivity extends AppCompatActivity implements PaymentStatusListener {

    private ImageView imageView;

    private TextView statusView;

    private Button payButton;

    private RadioGroup radioAppChoice;

    private EditText fieldPayeeVpa;
    private EditText fieldPayeeName;
    private EditText fieldTransactionId;
    private EditText fieldTransactionRefId;
    private EditText fieldDescription;
    private EditText fieldAmount;

    private EasyUpiPayment mEasyUpiPayment;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        initViews();

        payButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                pay();
            }
        });
    }

    private void initViews() {
        imageView = findViewById(R.id.imageView);
        statusView = findViewById(R.id.textView_status);
        payButton = findViewById(R.id.button_pay);

        fieldPayeeVpa = findViewById(R.id.field_vpa);
        fieldPayeeName = findViewById(R.id.field_name);
        fieldTransactionId = findViewById(R.id.field_transaction_id);
        fieldTransactionRefId = findViewById(R.id.field_transaction_ref_id);
        fieldDescription = findViewById(R.id.field_description);
        fieldAmount = findViewById(R.id.field_amount);

        String transactionId = "TID" + System.currentTimeMillis();
        fieldTransactionId.setText(transactionId);
        fieldTransactionRefId.setText(transactionId);

        radioAppChoice = findViewById(R.id.radioAppChoice);
    }

    private void pay() {
        String payeeVpa = fieldPayeeVpa.getText().toString();
        String payeeName = fieldPayeeName.getText().toString();
        String transactionId = fieldTransactionId.getText().toString();
        String transactionRefId = fieldTransactionRefId.getText().toString();
        String description = fieldDescription.getText().toString();
        String amount = fieldAmount.getText().toString();
        RadioButton paymentAppChoice = findViewById(radioAppChoice.getCheckedRadioButtonId());

        // START PAYMENT INITIALIZATION
        mEasyUpiPayment = new EasyUpiPayment.Builder()
                .with(this)
                .setPayeeVpa(payeeVpa)
                .setPayeeName(payeeName)
                .setTransactionId(transactionId)
                .setTransactionRefId(transactionRefId)
                .setDescription(description)
                .setAmount(amount)
                .build();

        // Register Listener for Events
        mEasyUpiPayment.setPaymentStatusListener(this);

        switch (paymentAppChoice.getId()) {
            case R.id.app_default:
                mEasyUpiPayment.setDefaultPaymentApp(PaymentApp.NONE);
                break;
            case R.id.app_amazonpay:
                mEasyUpiPayment.setDefaultPaymentApp(PaymentApp.AMAZON_PAY);
                break;
            case R.id.app_bhim_upi:
                mEasyUpiPayment.setDefaultPaymentApp(PaymentApp.BHIM_UPI);
                break;
            case R.id.app_google_pay:
                mEasyUpiPayment.setDefaultPaymentApp(PaymentApp.GOOGLE_PAY);
                break;
            case R.id.app_phonepe:
                mEasyUpiPayment.setDefaultPaymentApp(PaymentApp.PHONE_PE);
                break;
            case R.id.app_paytm:
                mEasyUpiPayment.setDefaultPaymentApp(PaymentApp.PAYTM);
                break;
        }

        // Check if app exists or not
        if (mEasyUpiPayment.isDefaultAppExist()) {
            onAppNotFound();
            return;
        }
        // END INITIALIZATION

        // START PAYMENT
        mEasyUpiPayment.startPayment();
    }

    @Override
    public void onTransactionCompleted(TransactionDetails transactionDetails) {
        // Transaction Completed
        Log.d("TransactionDetails", transactionDetails.toString());
        statusView.setText(transactionDetails.toString());
    }

    @Override
    public void onTransactionSuccess() {
        // Payment Success
        Toast.makeText(this, "Success", Toast.LENGTH_SHORT).show();
        imageView.setImageResource(R.drawable.ic_success);
    }

    @Override
    public void onTransactionSubmitted() {
        // Payment Pending
        Toast.makeText(this, "Pending | Submitted", Toast.LENGTH_SHORT).show();
        imageView.setImageResource(R.drawable.ic_success);
    }

    @Override
    public void onTransactionFailed() {
        // Payment Failed
        Toast.makeText(this, "Failed", Toast.LENGTH_SHORT).show();
        imageView.setImageResource(R.drawable.ic_failed);
    }

    @Override
    public void onTransactionCancelled() {
        // Payment Cancelled by User
        Toast.makeText(this, "Cancelled", Toast.LENGTH_SHORT).show();
        imageView.setImageResource(R.drawable.ic_failed);
    }

    @Override
    public void onAppNotFound() {
        Toast.makeText(this, "App Not Found", Toast.LENGTH_SHORT).show();
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        mEasyUpiPayment.detachListener();
    }
}

File: app/src/main/java/com/example/easyupipayment/kotlin/MainActivity.kt
package com.example.easyupipayment.kotlin

import android.os.Bundle
import android.util.Log
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.example.easyupipayment.R
import dev.shreyaspatil.easyupipayment.EasyUpiPayment
import dev.shreyaspatil.easyupipayment.listener.PaymentStatusListener
import dev.shreyaspatil.easyupipayment.model.PaymentApp
import dev.shreyaspatil.easyupipayment.model.TransactionDetails
import dev.shreyaspatil.easyupipayment.model.TransactionStatus
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity(), PaymentStatusListener {

	private lateinit var easyUpiPayment: EasyUpiPayment

	override fun onCreate(savedInstanceState: Bundle?) {
		super.onCreate(savedInstanceState)
		setContentView(R.layout.activity_main)

		initViews()
	}

	private fun initViews() {
		val transactionId = "TID" + System.currentTimeMillis()
		field_transaction_id.setText(transactionId)
		field_transaction_ref_id.setText(transactionId)

		// Setup click listener for Pay button
		button_pay.setOnClickListener { pay() }
	}

	private fun pay() {
		val payeeVpa = field_vpa.text.toString()
		val payeeName = field_name.text.toString()
		val transactionId = field_transaction_id.text.toString()
		val transactionRefId = field_transaction_ref_id.text.toString()
		val payeeMerchantCode = field_payee_merchant_code.text.toString()
		val description = field_description.text.toString()
		val amount = field_amount.text.toString()
		val paymentAppChoice = radioAppChoice

		val paymentApp = when (paymentAppChoice.checkedRadioButtonId) {
			R.id.app_default -> PaymentApp.ALL
			R.id.app_amazonpay -> PaymentApp.AMAZON_PAY
			R.id.app_bhim_upi -> PaymentApp.BHIM_UPI
			R.id.app_google_pay -> PaymentApp.GOOGLE_PAY
			R.id.app_phonepe -> PaymentApp.PHONE_PE
			R.id.app_paytm -> PaymentApp.PAYTM
			else -> throw IllegalStateException("Unexpected value: " + paymentAppChoice.id)
		}

		try {
			// START PAYMENT INITIALIZATION
			easyUpiPayment = EasyUpiPayment(this) {
				this.paymentApp = paymentApp
				this.payeeVpa = payeeVpa
				this.payeeName = payeeName
				this.transactionId = transactionId
				this.transactionRefId = transactionRefId
				this.payeeMerchantCode = payeeMerchantCode
				this.description = description
				this.amount = amount
			}
			// END INITIALIZATION

			// Register Listener for Events
			easyUpiPayment.setPaymentStatusListener(this)

			// Start payment / transaction
			easyUpiPayment.startPayment()
		} catch (e: Exception) {
			e.printStackTrace()
			toast("Error: ${e.message}")
		}
	}

	override fun onTransactionCompleted(transactionDetails: TransactionDetails) {
		// Transaction Completed
		Log.d("TransactionDetails", transactionDetails.toString())
		textView_status.text = transactionDetails.toString()

		when (transactionDetails.transactionStatus) {
			TransactionStatus.SUCCESS -> onTransactionSuccess()
			TransactionStatus.FAILURE -> onTransactionFailed()
			TransactionStatus.SUBMITTED -> onTransactionSubmitted()
		}
	}

	override fun onTransactionCancelled() {
		// Payment Cancelled by User
		toast("Cancelled by user")
		imageView.setImageResource(R.drawable.ic_failed)
	}

	private fun onTransactionSuccess() {
		// Payment Success
		toast("Success")
		imageView.setImageResource(R.drawable.ic_success)
	}

	private fun onTransactionSubmitted() {
		// Payment Pending
		toast("Pending | Submitted")
		imageView.setImageResource(R.drawable.ic_success)
	}

	private fun onTransactionFailed() {
		// Payment Failed
		toast("Failed")
		imageView.setImageResource(R.drawable.ic_failed)
	}

	private fun toast(message: String) {
		Toast.makeText(this, message, Toast.LENGTH_SHORT).show()
	}

//    Uncomment this if you have inherited [android.app.Activity] and not [androidx.appcompat.app.AppCompatActivity]
//
//	override fun onDestroy() {
//		super.onDestroy()
//		easyUpiPayment.removePaymentStatusListener()
//	}
}